lib/rules/named-grid-areas-no-invalid/index.js

```diff
@@ -0,0 +1,131 @@
+// @ts-nocheck
+
+'use strict';
+
+const _ = require('lodash');
+const report = require('../../utils/report');
+const ruleMessages = require('../../utils/ruleMessages');
+
+const ruleName = 'named-grid-areas-no-invalid';
+
+const messages = ruleMessages(ruleName, {
+	noRows: () => 'no rows defined',
+	notRectangular: () => 'template is not rectangular',
+	notContiguousOrRectangular: (areaNames) =>
+		`The following areas are not contiguous or rectangular ${areaNames.join(', ')}`,
+});
+
+function areasArray(gridTemplateString) {
+	const rows = Array.from(gridTemplateString.matchAll(/["']([^"']*)["']/g)).map(
```

Comment:
`.matchAll` is not supported by node version 10 in the test matrix. A new commit replaces this with `match` and a `while` loop. If there is a better way than this, let me know.